### PR TITLE
rebuild openssl-sys if the underlying openssl has changed

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

File: openssl-sys/build/main.rs [hunk 1]
```diff
@@ -79,6 +79,9 @@ fn main() {
     let target = env::var("TARGET").unwrap();
 
     let (lib_dirs, include_dir) = find_openssl(&target);
+    if let Some(printable_include) = include_dir.to_str() {
+        println!("cargo:rerun-if-changed={}", printable_include);
+    }
```
Comment (sfackler) [inline openssl-sys/build/main.rs:84]: I've considered doing something like this before, but I think it's a bit dangerous. With this specific setup, `include_dir` will be e.g. `/usr/include` on a system-wide install so you'd rebuild any time you updated any system header.

**Comment by alex** [comment]
`include_dir.join("openssl")`?

File: openssl-sys/build/main.rs (hunk 1 above)
Comment (reaperhulk) [reply openssl-sys/build/main.rs:86]: I've updated the PR to use `<include>/openssl`

File: openssl-sys/build/main.rs [hunk 2]
```diff
@@ -79,6 +79,11 @@ fn main() {
     let target = env::var("TARGET").unwrap();
 
     let (lib_dirs, include_dir) = find_openssl(&target);
+    let mut ossl_include_dir = include_dir.clone();
+    ossl_include_dir.push("openssl");
```
Comment (alex) [inline openssl-sys/build/main.rs:83]: This can just be `let ossl_include_dir = include_dir.join("openssl");` as far as I can tell

**Review by alex — APPROVED** [review]
(Not merging until @sfackler reviews)

**Review by sfackler — APPROVED** [review]

**Pull request merged**